adapters/syslog/syslog.go

```diff
@@ -81,104 +158,80 @@ func NewSyslogAdapter(route *router.Route) (router.LogAdapter, error) {
 	if err != nil {
 		return nil, err
 	}
+	connIsUDP := reflect.TypeOf(conn).String() == "*net.UDPConn"
 
-	format := cfg.GetEnvDefault("SYSLOG_FORMAT", "rfc5424")
-	priority := cfg.GetEnvDefault("SYSLOG_PRIORITY", "{{.Priority}}")
-	pid := cfg.GetEnvDefault("SYSLOG_PID", "{{.Container.State.Pid}}")
-	hostname = getHostname()
-
-	tag := cfg.GetEnvDefault("SYSLOG_TAG", "{{.ContainerName}}"+route.Options["append_tag"])
-	structuredData := cfg.GetEnvDefault("SYSLOG_STRUCTURED_DATA", "")
-	if route.Options["structured_data"] != "" {
-		structuredData = route.Options["structured_data"]
+	var format Format
+	if format, err = getFormat(); err != nil {
+		return nil, err
 	}
-	data := cfg.GetEnvDefault("SYSLOG_DATA", "{{.Data}}")
-	timestamp := cfg.GetEnvDefault("SYSLOG_TIMESTAMP", "{{.Timestamp}}")
+	debug("setting format to:", format)
 
-	if structuredData == "" {
-		structuredData = "-"
-	} else {
-		structuredData = fmt.Sprintf("[%s]", structuredData)
+	var tmpl *FieldTemplates
+	if tmpl, err = getFieldTemplates(route); err != nil {
+		return nil, err
 	}
```

Comment:
style nitpick: you could avoid extraneous var declaration thusly
```
	tmpl, err := getFieldTemplates(route)
	if err != nil {
		return nil, err
	}
```

Reply:
I initially implemented it like that but was getting type inference errors.
Will take a look again with your suggestion!  Thanks for the patience, I'm a newbie with GoLang :)

Reply:
No worries at all. Thanks for the help here and welcome to go! 😄